.github/workflows/unit_tests.yml

```diff
@@ -11,18 +11,25 @@ jobs:
   build:
     runs-on: ubuntu-22.04
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.10
-      uses: actions/setup-python@v2
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.8
```

Comment:
What breaks in 3.10?
Reverting to 3.8 makes us change some type annotations that we'll want to change again later, which isn't great.
But if supporting 3.10 is painful, we can go one step at a time - this is already an improvement.
Eventually, I'd like to support every supported Python version? Does that sound reasonable?

Reply:
Thank you for reviewing!

> What breaks in 3.10?

Actually nothing. The readme names 3.8 as minimal supported version, so I figured to set the version used in the ci accordingly.
I would also say that targeting at least the lowest version of Python that still receives security fixes would be a nice for users. 

> Reverting to 3.8 makes us change some type annotations that we'll want to change again later, which isn't great.

I think that this can be done with ease using ruff (or similar): [FA100](https://docs.astral.sh/ruff/rules/future-rewritable-type-annotation/ ), [FA102](https://docs.astral.sh/ruff/rules/future-required-type-annotation/)
This is also the first time that I came across the __future__ module usage for type hints. Maybe we should use these here?

I think that testing against multiple versions (in a matrix?) is something yet to be discussed in #399

Reply:
Yep okay; it makes sense to target Py38 then.

For the type annotations, and generally for the Py39+ features, we can use future but I think it's also reasonable to just write Python 3.8 code. Anyway, not a conversation specific to this CR.